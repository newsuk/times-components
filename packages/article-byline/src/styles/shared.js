import styleguide from "@times-components/styleguide";

const { colours, fontFactory } = styleguide();
const shared = {
  link: {
    color: colours.functional.action,
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    })
  },
  text: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    flexDirection: "row"
  }
};

export default shared;
