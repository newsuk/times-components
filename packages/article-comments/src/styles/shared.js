import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const styles = {
  button: {
    marginBottom: spacing(0),
    marginTop: spacing(5),
    maxWidth: 215
  },
  container: {
    alignItems: "center",
    borderStyle: "solid",
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    width: "100%"
  },
  headline: {
    color: colours.functional.primary,
    ...fontFactory({
      font: "headline",
      fontSize: "commentsHeadline"
    }),
    maxWidth: 315,
    paddingBottom: spacing(2),
    paddingTop: spacing(6),
    textAlign: "center"
  },
  link: {
    color: colours.functional.action
  },
  supporting: {
    ...fontFactory({
      font: "supporting",
      fontSize: "tertiary"
    }),
    color: colours.functional.secondary,
    maxWidth: 280,
    textAlign: "center"
  }
};

export default styles;
