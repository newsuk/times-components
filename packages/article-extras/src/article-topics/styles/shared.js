import { colours, tabletWidth } from "@times-components/styleguide";

const sharedStyles = {
  topicsContainer: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  topicsContainerTablet: {
    alignSelf: "center",
    width: tabletWidth
  },
  topicsMetaContainer: {
    justifyContent: "flex-start"
  }
};

export default sharedStyles;
