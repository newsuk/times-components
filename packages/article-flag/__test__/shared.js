/* eslint-env jest */

import "react-native";
import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";

import ArticleFlag, {
  NewArticleFlag,
  UpdatedArticleFlag,
  ExclusiveArticleFlag,
  SponsoredArticleFlag
} from "../article-flag";

module.exports = () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArticleFlag title="articleFlag" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders New flag correctly", () => {
    const tree = renderer.create(<NewArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Updated flag correctly", () => {
    const tree = renderer.create(<UpdatedArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Exclusive flag correctly", () => {
    const tree = renderer.create(<ExclusiveArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Sponsored flag correctly", () => {
    const tree = renderer.create(<SponsoredArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });
};
