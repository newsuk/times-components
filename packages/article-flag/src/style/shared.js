import styleguide from "@times-components/styleguide";

const { fontFactory } = styleguide();
const styles = {
  diamond: {
    marginRight: 5
  },
  title: {
    ...fontFactory({
      font: "bodyRegularSmallCaps",
      fontSize: "cardMetaMobile"
    }),
    fontWeight: "400",
    letterSpacing: 1.4
  },
  view: {
    alignItems: "center",
    flexDirection: "row"
  }
};

export default styles;
