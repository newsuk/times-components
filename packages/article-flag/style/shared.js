import { fonts } from "@times-components/styleguide";

const styles = {
  view: {
    flexDirection: "row",
    alignItems: "center"
  },
  diamond: {
    marginRight: 5,
    marginBottom: 1
  },
  title: {
    fontFamily: fonts.bodyRegularSmallCaps,
    fontSize: 12,
    fontWeight: "400",
    letterSpacing: 1.4
  }
};

export default styles;
