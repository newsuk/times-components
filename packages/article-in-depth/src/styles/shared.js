import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleHeadline: {
    ...fontFactory({
      font: "headline",
      fontSize: "headline"
    }),
    color: colours.functional.brandColour,
    marginBottom: spacing(2),
    textAlign: "center"
  },
  container: {
    alignItems: "center",
    paddingBottom: spacing(9),
    paddingHorizontal: spacing(4),
    paddingTop: spacing(12)
  },
  datePublication: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    marginTop: spacing(1)
  },
  flag: {
    flexDirection: "row",
    marginBottom: spacing(3),
    marginTop: spacing(1)
  },
  flagContainer: {
    marginRight: spacing(3)
  },
  label: {
    marginBottom: spacing(2)
  },
  leadAssetContainer: {
    marginBottom: spacing(4)
  },
  meta: {
    flexDirection: "row",
    flexWrap: "wrap"
  },
  metaContainer: {
    alignItems: "center",
    borderBottomWidth: 1,
    borderColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    marginHorizontal: spacing(2),
    paddingBottom: 6,
    paddingTop: spacing(2)
  },
  standFirst: {
    ...fontFactory({
      font: "headlineRegular",
      fontSize: "smallestHeadline"
    }),
    color: colours.functional.primary,
    marginBottom: spacing(3),
    paddingHorizontal: spacing(2),
    textAlign: "center"
  }
};

export default sharedStyles;
