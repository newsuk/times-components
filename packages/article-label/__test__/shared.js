/* eslint-env jest */

import "react-native";
import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import ArticleLabel from "../article-label";

module.exports = () => {
  it("renders ArticleLabel", () => {
    const tree = renderer
      .create(<ArticleLabel title="swimming" color="#008347" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
};
