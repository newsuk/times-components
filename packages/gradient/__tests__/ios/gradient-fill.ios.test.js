import shared from "../shared-fill.native";

describe("ios", () => {
  shared();
});
