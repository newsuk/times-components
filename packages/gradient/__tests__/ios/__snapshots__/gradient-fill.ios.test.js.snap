// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. gradient with no angle 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          -0,
          0,
          0.9764705882352941,
          0.9764705882352941,
          0.9764705882352941,
          1,
          0.9294117647058824,
          0.9294117647058824,
          0.9294117647058824,
          1,
          0,
          1,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`2. gradient with an angle (-45) 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          -0,
          0,
          0.9764705882352941,
          0.9764705882352941,
          0.9764705882352941,
          1,
          0.9294117647058824,
          0.9294117647058824,
          0.9294117647058824,
          1,
          0,
          1,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`3. gradient with an angle (45) 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          0,
          -0,
          0.9764705882352941,
          0.9764705882352941,
          0.9764705882352941,
          1,
          0.9294117647058824,
          0.9294117647058824,
          0.9294117647058824,
          1,
          0,
          1,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`4. gradient with an angle (90) 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          0,
          0,
          0.9764705882352941,
          0.9764705882352941,
          0.9764705882352941,
          1,
          0.9294117647058824,
          0.9294117647058824,
          0.9294117647058824,
          1,
          0,
          1,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`5. gradient with an angle (180) 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          0,
          0,
          0.9764705882352941,
          0.9764705882352941,
          0.9764705882352941,
          1,
          0.9294117647058824,
          0.9294117647058824,
          0.9294117647058824,
          1,
          0,
          1,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`6. gradient with an angle (270) 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          0,
          0,
          0.9764705882352941,
          0.9764705882352941,
          0.9764705882352941,
          1,
          0.9294117647058824,
          0.9294117647058824,
          0.9294117647058824,
          1,
          0,
          1,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`7. gradient with custom colours 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          -0,
          0,
          1,
          0,
          0,
          1,
          0,
          1,
          1,
          1,
          0,
          1,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`8. gradient with custom points 1`] = `
<View>
  <ARTSurfaceView>
    <ARTShape
      fill={
        Array [
          1,
          0,
          0,
          -0,
          0,
          0.9294117647058824,
          0.9294117647058824,
          0.9294117647058824,
          1,
          0.9764705882352941,
          0.9764705882352941,
          0.9764705882352941,
          1,
          0.8,
          0.2,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;
